src/ggrc/assets/javascripts/components/assessment_attributes.js

```
@@ -81,62 +81,62 @@
          */
         removeField: function (scope, el, ev) {
           ev.preventDefault();
           scope.attr('_pending_delete', true);
         },
         /*
-         * Split field values
-         *
-         * @return {Array} attrs - Returns split values as an array
-         */
-        attrs: function () {
-          if (_.contains(['Person', 'Text'], this.field.attr('type'))) {
-            return [this.attr('field.multi_choice_options')];
-          }
-          return _.splitTrim(this.attr('field.multi_choice_options'), {
-            compact: true
-          });
-        },
-        /*
-         * Denormalize field.multi_choice_mandatory into field.opts
+         * Denormalize field.multi_choice_mandatory into opts
          * "0, 1, 2" is normalized into
-         * { attachment-0: false, comment-0: false,
-         *   attachment-1: false, comment-1: true,
-         *   attachment-2: true, comment-2: false
-         * }
+         * [
+         * {value: 0, attachment: false, comment: false},
+         * {value: 1, attachment: false, comment: true},
+         * {value: 2, attachment: true, comment: false},
+         * ]
          */
         denormalize_mandatory: function (field, pads) {
           var options = _.splitTrim(field.attr('multi_choice_options'));
           var vals = _.splitTrim(field.attr('multi_choice_mandatory'));
-          var mand = new can.Map();
-          _.each(_.zip(options, vals), function (zip) {
-            var option = zip[0];
+          var isEqualLength = options.length === vals.length;
+          var range;
+
+          if (!isEqualLength && options.length < vals.length) {
+            vals.length = options.length;
+          } else if (!isEqualLength && options.length > vals.length) {
+            range = _.range(options.length - vals.length);
+            range = range.map(function () {
+              return '0';
+            });
+            vals = vals.concat(range);
+          }
```

Hm... this equalizes the length of both lists, but are we sure this is the best way to handle it? My concern here is that we only fix a symptom here, not the cause itself, i.e. why the list lengths differ in the first place, indicating a bug in the `field` object's values.

As suggested in the [initial comment](https://github.com/google/ggrc-core/pull/4693#discussion_r88289745), it would probably be better to throw an error here, so that are notified about a potential deeper issue that might cause bugs elsewhere, too.